sdk/communication/Azure.Communication.CallingServer/src/ConversationClient.cs

```diff
@@ -0,0 +1,639 @@
+        /// <exception cref="ArgumentNullException"><paramref name="conversationId"/> is null.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="callOptions"/> is null.</exception>
+        public virtual async Task<Response<JoinCallResponse>> JoinCallAsync(string conversationId, CommunicationIdentifier source, JoinCallOptions callOptions, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(CallClient)}.{nameof(JoinCall)}");
+            scope.Start();
+            try
+            {
+                Argument.AssertNotNull(callOptions, nameof(callOptions));
+
+                return await RestClient.JoinCallAsync(
+                    conversationId: conversationId,
+                    source: CommunicationIdentifierSerializer.Serialize(source),
+                    callbackUri: callOptions.CallbackUri?.AbsoluteUri,
+                    requestedModalities: callOptions.RequestedModalities,
+                    requestedCallEvents: callOptions.RequestedCallEvents,
+                    subject: null,
+                    cancellationToken: cancellationToken
+                    ).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// Join the call using conversation id.
+        /// <param name="conversationId"> The conversation id that can be a group id or a encoded conversation url retrieve from client. </param>
+        /// <param name="source"> The source identity. </param>
+        /// <param name="callOptions"> The call Options. </param>
+        /// <param name="cancellationToken"> The cancellation token. </param>
+        /// <exception cref="RequestFailedException">The server returned an error. See <see cref="Exception.Message"/> for details returned from the server.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is null.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="conversationId"/> is null.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="callOptions"/> is null.</exception>
+        public virtual Response<JoinCallResponse> JoinCall(string conversationId, CommunicationIdentifier source, JoinCallOptions callOptions, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(CallClient)}.{nameof(JoinCall)}");
+            scope.Start();
+            try
+            {
+                Argument.AssertNotNull(callOptions, nameof(callOptions));
+
+                return RestClient.JoinCall(
+                    conversationId: conversationId,
+                    source: CommunicationIdentifierSerializer.Serialize(source),
```

Comment:
Do you also need to check that the source is not null since you are serializing it?

Reply:
this will get addressed in the update.